File: 백준/Silver/2156. 포도주 시식/포도주 시식.cc
#include <iostream>
#include <algorithm>
using namespace std;

int wine[10001] = {0,};
int dp[10001] = {0,};

int main() {
    int n;
    
    cin >> n;
    
    for(int i = 1; i <= n; i++) {
        cin >> wine[i];
    }
    
    for(int i = 1; i <= n; i++) {
        dp[i] = max(dp[i-2] + wine[i], max(dp[i-3] + wine[i-1] + wine[i], dp[i-1]));
    }
    
    cout << dp[n];
}

File: 백준/Silver/2156. 포도주 시식/포도주 시식.swift
let n = Int(readLine()!)!
var wine = [Int]()
var drink = Array(repeating: 0, count: 10001)

wine.append(0)

for _ in 1...n {
    wine.append(Int(readLine()!)!)
}

drink[1] = wine[1]

if n >= 2 {
    drink[2] = wine[1] + wine[2]
}

if n >= 3 {
    for i in 3...n {
        drink[i] = max(drink[i - 2] + wine[i], drink[i - 3] + wine[i - 1] + wine[i], drink[i - 1])
    }
}

print(drink[n])